util/fibers/epoll_socket.cc

```diff
@@ -470,66 +466,72 @@ void EpollSocket::ResetOnRecvHook() {
   }
 }
 
-void EpollSocket::HandleAsyncRequest(error_code ec, bool is_send) {
-  uint8_t async_pending = is_send ? async_write_pending_ : async_read_pending_;
-  if (async_pending) {
-    auto& async_request = is_send ? async_write_req_ : async_read_req_;
-    DCHECK(async_request);
-
-    auto finalize_and_fetch_cb = [this, &async_request, is_send]() {
-      auto cb = std::move(async_request->cb);
-      delete async_request;
-      async_request = nullptr;
-      if (is_send)
-        async_write_pending_ = 0;
-      else
-        async_read_pending_ = 0;
-      return cb;
-    };
-
-    if (ec) {
-      auto cb = finalize_and_fetch_cb();
-      cb(make_unexpected(ec));
-    } else if (auto res = async_request->Run(native_handle(), is_send); res.first) {
-      auto cb = finalize_and_fetch_cb();
-      cb(res.second);
-    }
-  } else if (recv_hook_registered_ && !is_send && !ec) {
-    on_recv_->cb(RecvNotification{});
-  } else {
-    auto& sync_request = is_send ? write_req_ : read_req_;
-    // It could be that we activated context already, but has not switched to it yet.
-    // Meanwhile a new event has arrived that triggered this callback again.
-    if (sync_request && sync_request->IsSuspended()) {
-      DVSOCK(2) << "Wakey: Schedule read in " << sync_request->name();
-      sync_request->Activate(ec);
+void EpollSocket::HandleAsyncRequest(unsigned epoll_mask) {
+  uint8_t masks[2] = {EpollProactor::EPOLL_IN, EpollProactor::EPOLL_OUT};
+
+  for (unsigned i = 0; i < 2; ++i) {
+    if ((epoll_mask & (masks[i] | kErrMask)) == 0) {
+      continue;
     }
-  }
-}
 
-void EpollSocket::Wakey(uint32_t ev_mask, int error, EpollProactor* cntr) {
-  DVSOCK(2) << "Wakey " << ev_mask;
-#ifdef __linux__
-  constexpr uint32_t kErrMask = EPOLLERR | EPOLLHUP | EPOLLRDHUP;
-#else
-  constexpr uint32_t kErrMask = POLLERR | POLLHUP;
-#endif
+    bool is_send = (i == 1);
```

Comment:
I see diffs here are hard. I can split this PR into two commits

Reply:
It's a small change, it's ok